keps/sig-auth/20190116-service-account-external-signing.md

```diff
@@ -127,9 +159,43 @@ New token generation and validation could suffer a performance difference when r
 
 Signing and verifying tokens over a grpc API carries the risk of a server side request forgery, where a malicious client could generate tokens. To mitigate this risk, the API will only be accessible over a unix socket.
 
-## Graduation Criteria
+## Design Details
+
+### Test Plan
+
+Unit tests covering:
+* ProjectedToken signing works with external signer client
+* OIDC keys URL shows public keys from external signer
+
+Integration tests covering:
+* ProjectedToken signing works with simple signer server
+* OIDC keys URL shows public keys from external signer
+* Validate apiserver flag logic (require issuer to be set)
+
+E2E test covering:
```

Comment:
Some thoughts:

- Tests that show that the infra keeps working when the external signer is deliberately made to be slow
- Tests that show that the infra mostly keeps working when the external signer is down